Create Security Policy

diogoteles08 opened the pull request:
Closes #3151

I've created the SECURITY.md file considering the [report vulnerability through security advisory](https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing/privately-reporting-a-security-vulnerability), which is a handy new GitHub feature, but it's still in beta and **has to be enabled**.

If you're interested in this feature, it can be activated by:
1. Clicking on this [link](https://github.com/HDFGroup/hdf5/settings/security_analysis) or go to Code security & analysis section on your repo's settings.
2. Clicking "Enable" for "Private vulnerability reporting (Beta)"

If you rather not use this feature, you can also request users to report vulnerabilities to an email. If that's the case, let me know which email it would be and I can submit the change.

Additionally, feel free to edit or suggest any changes to this document, it is supposed to reflect the amount of effort the team can offer to handle vulnerabilities.

diogoteles08 committed:
Create Security Policy

derobins reviewed (approved):
I'll go ahead and approve this, even though it's just a skeleton. I've been meaning to add a real security policy for a while now. Guess this is my excuse to do so!

diogoteles08 commented:
Sounds great! Let me know if I can be of any help =)

diogoteles08 commented:
Oh and FWI, the GitHub's feature of private vulnerability reporting was not yet enabled for your repository, so the link I added on the Security Policy is reaching 404. 

I know it's just a skeleton and you will rework the policy, but if you want to enable the security advisory by now, I've added the instructions on the body of this PR.

derobins commented:
Cool. Thanks! I turned that on.
